crates/sui-replay/src/batch_replay.rs

```diff
@@ -0,0 +1,159 @@
+// Copyright (c) Mysten Labs, Inc.
+// SPDX-License-Identifier: Apache-2.0
+
+use crate::replay::LocalExec;
+use crate::types::ReplayEngineError;
+use futures::future::join_all;
+use parking_lot::Mutex;
+use std::collections::VecDeque;
+use std::sync::atomic::{AtomicBool, AtomicU64};
+use std::sync::Arc;
+use sui_config::node::ExpensiveSafetyCheckConfig;
+use sui_types::base_types::TransactionDigest;
+use tokio::time::Instant;
+use tracing::{error, info};
+
+pub async fn batch_replay(
+    tx_digests: impl Iterator<Item = TransactionDigest>,
+    batch_size: u64,
+    rpc_url: String,
+    expensive_safety_check_config: ExpensiveSafetyCheckConfig,
+    use_authority: bool,
+    terminate_early: bool,
+) {
+    let provider = Arc::new(TransactionDigestProvider::new(tx_digests));
+    let mut tasks = vec![];
+    let cur_time = Instant::now();
+    for _ in 0..batch_size {
+        let provider = provider.clone();
+        let expensive_safety_check_config = expensive_safety_check_config.clone();
+        let rpc_url_ref = rpc_url.as_str();
+        tasks.push(async move {
+            run_task(
+                provider,
+                rpc_url_ref,
+                expensive_safety_check_config,
+                use_authority,
+                terminate_early,
+            )
+            .await
+        });
```

Comment:
Asking for my own education: Is there a difference between what's written here, and this:

```suggestion
        tasks.push(run_task(
            provider,
            rpc_url_ref,
            expensive_safety_check_config,
            use_authority,
            terminate_early,
        ));
```

?

Reply:
Good question. I think they are functionally identical in this case. What you suggested is probably more efficient.